lightning/src/routing/router.rs

```diff
@@ -1286,10 +1321,32 @@ pub fn get_route<L: Deref>(our_node_pubkey: &PublicKey, network: &NetworkGraph,
 	Ok(route)
 }
 
+struct FeePenaltyCache<'a, S: routing::Score> {
```

Comment:
Actually, if I remove the cache entirely I get a substantial speedup. I think that makes sense - for a given pass through the graph, we really shouldn't query twice for a given channel - dijkstra's should basically just hit it once.

```
test routing::router::benches::generate_mpp_routes        ... bench:  44,775,553 ns/iter (+/- 25,442,682)
test routing::router::benches::generate_routes            ... bench:  44,087,224 ns/iter (+/- 30,301,992)
```
to
```
test routing::router::benches::generate_mpp_routes        ... bench:  37,863,649 ns/iter (+/- 21,238,953)
test routing::router::benches::generate_routes            ... bench:  38,099,581 ns/iter (+/- 25,742,071)
```